### ISSUE TITLE
broken on SmartOS

### ISSUE THREAD

**Issue by robinbowes** [saltstack/salt-bootstrap#224]
I tried using salt-bootstrap on SmartOS with pkgin configured and found that it's broken.

I dug in and found a few issues:
1. There is no gcc-compiler package. Not sure why this is missing, or if it ever existed (presumably did?)
2. logic error when installing gcc47. This:

```
pkg_info gcc47 > /dev/null 2>&1 && pkgin -y in gcc47
```

should be:

```
pkg_info gcc47 > /dev/null 2>&1 || pkgin -y in gcc47
```
1. gcc47 doesn't leave gcc in the path, it installs to /opt/local/gcc47/bin/gcc
2. I added /opt/local/gcc47/bin/ to the path but the zeromq build fails with:

```
Making all in src
make[1]: Entering directory `/root/zeromq-3.2.3/src'
make  all-am
make[2]: Entering directory `/root/zeromq-3.2.3/src'
  CXX    libzmq_la-xpub.lo
make[2]: *** [libzmq_la-xpub.lo] Error 1
make[2]: Leaving directory `/root/zeromq-3.2.3/src'
make[1]: *** [all] Error 2
make[1]: Leaving directory `/root/zeromq-3.2.3/src'
make: *** [all-recursive] Error 1
 * ERROR: Failed to run install_smartos_git_deps()!!!
```

I noticed that there is a salt package in pkgin but it is 0.15.3. Might be worth getting that updated? I believe the maintainer is Jonathan Perkin: http://www.perkin.org.uk/pages/pkgsrc-binary-packages-for-illumos.html

R.

**Comment by techhat** [comment]
Hi @robinbowes. Indeed there was a gcc-compiler package at one point. My first thought is that either the system that you're on isn't pointing to a repo with gcc-compiler, or it is but the package was removed from the repo. Do you know which pkgin repo your system is pointed to?

**Comment by robinbowes** [comment]
I'm using the 2013Q2 repo:

http://pkgsrc.joyent.com/packages/SmartOS/2013Q2/x86_64/All

R.

**Comment by techhat** [comment]
Thanks, @robinbowes, I'll take a look at it. Is it safe to assume this is a VM on JoyEnt? If so, can you tell me which image you spun up?

**Comment by robinbowes** [comment]
Jo,

I am trying to install salt in the global zone of a SmartOS machine, version joyent_20130905T204057Z

Actually, I only need to install the minion.

R.

**Comment by mguegan** [comment]
Hi ! 
Since 2013Q2 is available, it's no more necessary to compile some packages with gcc. Currently I'm using pkgin to install salt directly, I've done some modification on my github repo, but it would require more testing before merging.
You can see the whole thing here : 

https://github.com/virtua-network/salt-bootstrap/blob/develop/bootstrap-salt.sh

Beware that I'm using a custom repo (with 0.16.3 instead of 2013Q2's 0.15 version)

**Comment by mguegan** [comment]
Please, can you try with this version ?

https://github.com/virtua-network/salt-bootstrap/blob/fix_smartos/bootstrap-salt.sh

The stable install seems to work.

You will need 2013Q2 pkgin on your GZ.

**Comment by s0undt3ch** [comment]
Fixed by #239
